### PR TITLE
Detect build/ correctly

### LINKED ISSUE #1091: k8s version sometimes not filled in during PR builds
Seen in https://github.com/kubernetes/kubernetes/pull/36883 (https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/36883/pull-kubernetes-e2e-kops-aws/20/):

```
W1116 12:38:36.666] + export 'E2E_OPT=  --kops-cluster aws-kops-agent-pr-47-0.test-aws.k8s.io  --kops-kubernetes-version https://storage.googleapis.com/kubernetes-release-pull/ci/pull-kubernetes-e2e-kops-aws/  --kops-nodes 4  --kops-state s3://k8s-kops-jenkins/'
```

### PR DESCRIPTION
**Pull request by zmerlynn**
Fixes #1091

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1093)

### CODE REVIEW

File: jobs/pull-kubernetes-e2e-kops-aws.sh [hunk 1]
```diff
@@ -38,7 +38,17 @@ if [[ -z "${SKIP_BUILD:-}" ]]; then
   ./hack/jenkins/build.sh
 fi
 
-readonly version=$(source build-tools/util.sh && echo $(kube::release::semantic_version))
+# TODO(spxtr): once we no longer have any PRs against release-1.4,
```
Comment (ixdy) [inline jobs/pull-kubernetes-e2e-kops-aws.sh]: the plan is for `build-tools/` to revert back to `build/`, actually.

File: jobs/pull-kubernetes-e2e-kops-aws.sh [hunk 2]
```diff
@@ -38,7 +38,17 @@ if [[ -z "${SKIP_BUILD:-}" ]]; then
...
+# remove the first branch.
+if [[ -e build/util.sh ]]; then
+  readonly version=$(source build/util.sh && echo $(kube::release::semantic_version))
+elif [[ -e build-tools/util.sh ]]; then
+  readonly version=$(source build-tools/util.sh && echo $(kube::release::semantic_version))
+fi
+if [[ -z "{version:-}" ]]; then
```
Comment (ixdy) [inline jobs/pull-kubernetes-e2e-kops-aws.sh]: missing `$`

Comment (zmerlynn) [inline jobs/pull-kubernetes-e2e-kops-aws.sh]: Ha. Flying fingers, sorry.

File: jobs/pull-kubernetes-e2e-kops-aws.sh (hunk 1 above)
Comment (zmerlynn) [inline jobs/pull-kubernetes-e2e-kops-aws.sh]: Changed comment

**Comment by zmerlynn** [comment]
PTAL

**Pull request merged**